### ISSUE TITLE
edges() and get_edgelist() redundant?

### ISSUE THREAD

**Issue by szhorvat** [igraph/igraph#2206]
There are two functions to get the endpoints of edges:

```
igraph_error_t igraph_edges(const igraph_t *graph, igraph_es_t eids, igraph_vector_int_t *edges);

igraph_error_t igraph_get_edgelist(const igraph_t *graph, igraph_vector_int_t *res, igraph_bool_t bycol);
```

The first can handle a subset of edges (the second can't), and the second has the `bycol` argument (the first doesn't). It seems to me that it makes sense to merge these two as part of the API cleanup.

One might say that `igraph_get_edgelist()` is matrix-oriented, so it should be changes to return a matrix. I am not sure that is necessary. If that matrix would only be read, but not modified, then one can always use `igraph_matrix_view_from_vector()`.

**Comment by szhorvat** [comment]
Regarding the `bycol` argument, this is used in the R interface.

Within the C/igraph codebase, there is one single place where it is set to `true`, in `igraph_isomorphic_bliss()`. Here it is used to get separate `to`/`from` vectors, to which the canonical permutation returned by Bliss can be applied. Thus, `get_edgelist()`  with `bycol=true` is not actually the perfect fit for the job here. That leaves the R interface as the only used of this feature.

**Comment by ntamas** [comment]
The Python interface could also start using this feature if we want to achieve zero-copy data transfer between the edge list returned from `igraph_get_edgelist()` and a NumPy array, depending on what ordering the NumPy array uses for storing the items (Fortran vs C conventions). Or, any other higher-level interface could benefit from the `bycol` argument for similar reasons. If we want to merge the two functions, we can, but I'd probably vote for keeping the `bycol` argument to make it possible to do zero-copy data transfers to a host language no matter what conventions the host language uses for arrays.

**Comment by szhorvat** [comment]
Yes, it makes sense. Merge and have both arguments: and edge selector and bycol. Sounds good?

**Comment by szhorvat** [comment]
Consolidate into single function with both `eids` and `bycol` parameters. Make change for 1.0.

**Comment by ntamas** [comment]
Which one should be consolidated into the other one and which one should be deprecated?

**Comment by szhorvat** [comment]
It's a naming choice, and I have no strong preference.

`igraph_get_edgelist()` is used more (21 vs 4 hits in my IDE).

The `get` naming is consistent with e.g. `get_adjacency()`

So I cast a soft vote for `igraph_get_edgelist()`.

**Comment by ntamas** [comment]
On the other hand: 

- `igraph_get_edgelist()` does not return the edge list of the graph if the edge selector that you pass to it refers to a subset of the graph only.
- `igraph_edges()` is currently part of the "core interface" of the graph data type, while `igraph_get_edgelist()` is a derived function, indicating that `igraph_edges()` is probably the one that is "more basic".

To ease the migration path for new users, I'm inclined to simply add the `bycol` argument to `igraph_edges()` and then rewrite `igraph_get_edgelist()` in terms of `igraph_edges()`. We can then keep `igraph_get_edgelist()` in 1.x with a deprecation warning, and remove it in 2.x if that time ever comes.

**Comment by ntamas** [comment]
In the end I decided to add a `bycol` argument to `igraph_edges()` but to keep `igraph_get_edgelist()` without deprecation. Getting the entire edge list is a very common operation and it becomes cumbersome if you always need to specify `igraph_ess_all(IGRAPH_EDGEORDER_ID)` in a call to `igraph_edges()`.
